vcd/resource_vcd_vapp_vm_hw_virtualization_test.go

```diff
@@ -0,0 +1,86 @@
+package vcd
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func TestAccVcdVAppVm_HardwareVirtualization(t *testing.T) {
+	vappNameHwVirt := "TestAccVcdVAppHwVirt"
+	vmNameHwVirt := "TestAccVcdVAppHwVirt"
+	var vapp govcd.VApp
+	var vm govcd.VM
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"Vdc":         testConfig.VCD.Vdc,
+		"EdgeGateway": testConfig.Networking.EdgeGateway,
+		"NetworkName": "TestAccVcdVAppVmNetHwVirt",
+		"Catalog":     testSuiteCatalogName,
+		"CatalogItem": testSuiteCatalogOVAItem,
+		"VappName":    vappNameHwVirt,
+		"VmName":      vmNameHwVirt,
+	}
+
+	configTextStep0 := templateFill(testAccCheckVcdVAppVm_hardwareVirtualization, params)
+
+	params["ExposeHardwareVirtualization"] = true
+	params["FuncName"] = t.Name() + "-step1"
+	configTextStep1 := templateFill(testAccCheckVcdVAppVm_hardwareVirtualization, params)
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configTextStep0)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVcdVAppVmDestroy(vappNameHwVirt),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configTextStep0,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVAppVmExists(vappNameHwVirt, vmNameHwVirt, "vcd_vapp_vm."+vmNameHwVirt, &vapp, &vm),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm."+vmNameHwVirt, "name", vmNameHwVirt),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm."+vmNameHwVirt, "expose_hardware_virtualization", "false"),
+				),
+			},
+			resource.TestStep{
+				Config: configTextStep1,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVAppVmExists(vappNameHwVirt, vmNameHwVirt, "vcd_vapp_vm."+vmNameHwVirt, &vapp, &vm),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm."+vmNameHwVirt, "name", vmNameHwVirt),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm."+vmNameHwVirt, "expose_hardware_virtualization", "true"),
+				),
+			},
+		},
+	})
+}
+
+const testAccCheckVcdVAppVm_hardwareVirtualization = `
+resource "vcd_vapp" "{{.VappName}}" {
+  name = "{{.VappName}}"
+  org  = "{{.Org}}"
+  vdc  = "{{.Vdc}}"
+}
+
+resource "vcd_vapp_vm" "{{.VmName}}" {
```

Comment:
This sample is not correctly formatted for Terraform

Reply:
Not sure I follow you. Could you be more specific?

Reply:
I've applied terraform fmt on a test snippet.